x-pack/plugins/aiops/public/embeddable/embeddable_chart_component_wrapper.tsx

```diff
@@ -170,7 +170,7 @@ export const ChartGridEmbeddableWrapper: FC<
       },
     });
 
-    if (partitions && fieldConfig.splitField) {
+    if (Array.isArray(partitions) && partitions.length > 0 && fieldConfig.splitField) {
```

Comment:
Fixes filter query if `partitions` field is defined as an empty array.

Reply:
Should the dashboard modal remember the selected partitions. This is always showing as blank when I edit the embeddable, even if I've selected individual partitions inside the ML AIOps page, or edited inside Dashboard.

<img width="485" alt="image" src="https://github.com/elastic/kibana/assets/7405507/3485632f-6bcc-4057-9f5c-fdbbbfeea4d0">

Reply:
Fixed in https://github.com/elastic/kibana/pull/169962/commits/6e121ce8ecd0c7b90c31773b1211a90d77e316b7. 
Both bugs with the form where related to `usePrevious` hook 🤔 It used to work correctly, very odd.